test/functional/test_framework/wallet.py

```diff
@@ -55,6 +55,15 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
+    def prepare_tx(self, utxo_to_spend, send_value):
```

Comment:
See this: https://github.com/bitcoin/bitcoin/pull/20808#discussion_r554610229

Basically I'm wondering if you're going to even need to touch `MiniWallet` in this PR if we can align here. If we extend `MiniWallet` to have an option to "just-create-don't-send" then you could
1) create the base tx with `MiniWallet` but don't send
2) add all the `big_txouts` to that `tx`
3) send it yourself, outside of `MiniWallet`

To me it seems like that may be the cleanest solution

Reply:
🤔 hmm i'll look into it and try to find a clean solution

Reply:
If I take this approach the code would be a lot cleaner but i'd have to get rid of some of the assertion checks for the tx because, i dont think i'll have access to variables such as `send_value` or `fee` unless I really dig into the `tx` list: 

<details open>
<summary>This code...</summary>
 
  <pre>minwallet._utxos.append({'txid': txid, 'vout': 0, 'value': send_value})  
tx_info = node.getmempoolentry(txid)
assert_equal(tx_info['fee'], fee)
large_txids.append({'txid': txid, 'wtxid': tx_info['wtxid'], 'hex': tx_hex})  </pre>
 
</details>  

The assertions aren't used in the current code. So i'm tempted to not include them in the new code unless they modify `MiniWallet` variable behaviors.

Reply:
I think it's normal to do some assertions outside of MiniWallet if you think they're needed. ie we have things in the current miniwallet tests like: `assert_equal(entry_time, node.getmempoolentry(parent_txid)['time'])` after sending a transaction.

Anyways do what you think is best here. If my proposed solution ends up having problems or doesn't make sense I don't want to mislead you

Reply:
For sure! pushed a new commit with required changes :)